plugins/auto_merge/merge_checks.go

```diff
@@ -48,13 +54,21 @@ func (plugin autoMergePlugin) checkMergeRequest(project *gitlab.Project, mergeRe
 	return status
 }
 
-func (plugin autoMergePlugin) mergeChecks() [6]mergeCheck {
-	return [...]mergeCheck{
-		checks.HasEnoughApprovalsCheck{},
+func (plugin autoMergePlugin) setupMergeChecks() {
+	if mergeChecks != nil {
+		return
+	}
+
+	mergeChecks = []mergeCheck{
+		checks.HasEnoughApprovalsCheck{
+			Client: plugin.Client,
```

Comment:
Why do some checks need a client and others do not?

Reply:
Some can simply get their data from the merge-request. others need to call the api again to get some additional data.